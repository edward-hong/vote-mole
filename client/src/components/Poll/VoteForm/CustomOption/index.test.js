import React from 'react'
import { render } from '@testing-library/react'

import CustomOption from '.'

describe('CustomOption', () => {
	it('shows error message when there is an error', () => {
		const error = 'Please enter your custom selection'
		const { getByText } = render(
			<CustomOption
				input={{ name: 'customOption' }}
				meta={{ touched: true, error }}
			/>,
		)

		expect(getByText(error)).toHaveTextContent(error)
	})

	it('renders correctly when valid', () => {
		const { container } = render(
			<CustomOption
				input={{ name: 'customOption' }}
				meta={{ touched: true, error: false }}
			/>,
		)

		expect(container.firstChild).toMatchSnapshot()
	})

	it('renders correctly when untouched', () => {
		const { container } = render(
			<CustomOption
				input={{ name: 'customOption' }}
				meta={{ touched: false, error: false }}
			/>,
		)

		expect(container.firstChild).toMatchSnapshot()
	})
})
