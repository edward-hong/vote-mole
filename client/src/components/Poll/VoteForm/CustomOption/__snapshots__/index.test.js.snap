// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomOption renders correctly when untouched 1`] = `
<div
  class="form-group"
>
  <label
    class=""
    for="customOption"
  />
  <input
    class="form-control"
    name="customOption"
    type="text"
  />
</div>
`;

exports[`CustomOption renders correctly when valid 1`] = `
<div
  class="form-group"
>
  <label
    class=""
    for="customOption"
  />
  <input
    class="is-valid form-control"
    name="customOption"
    type="text"
  />
</div>
`;
